### DISCUSSION TITLE
Analysing previous days

### DISCUSSION THREAD

**Discussion by phillprice** [mcguirepr89/BirdNET-Pi#543] [Q&A] [answered]
For some reason my device recorded fine all of Saturday but the analysis fell over.

When I got it restarted it picked up from Sunday so all of Saturday's recordings are sat untouched. 

I've tried manually amending the `analyzing_now.txt` to a file from Saturday and I've also tried running this from the command line on the pi which gave the error below. Is there any chance of picking those files up or are they lost.

```
BirdNET-Pi/birdnet/bin/python3 BirdNET-Pi/scripts/analyze.py --i BirdSongs/September-2022/10-Saturday/2022-09-10-birdnet-00:02:22.wav --o BirdSongs/September-2022/10-Saturday/2022-09-10-birdnet-00:02:22.wav.csv --lat <LAT> --lon <LON> --week 34 --overlap 0.0 --sensitivity 1.25 --min_conf 0.7 --exclude_list BirdNET-Pi/exclude_species_list.txt --birdweather_id "<ID>"
```
Error:
```
Traceback (most recent call last):
  File "/home/pi/BirdNET-Pi/scripts/analyze.py", line 117, in <module>
    main()
  File "/home/pi/BirdNET-Pi/scripts/analyze.py", line 108, in main
    send(DISCONNECT_MESSAGE)
  File "/home/pi/BirdNET-Pi/scripts/analyze.py", line 21, in send
    client.send(message)
BrokenPipeError: [Errno 32] Broken pipe
```

**Comment by mcguirepr89** [comment, marked as answer]
Hello, @phillprice -- I'm sorry BirdNET-Pi crapped out on you Saturday, that's a bummer. 

If the raw data is still there, we ought to be able to tinker the system to fetch those files, though it [does that already](https://github.com/mcguirepr89/BirdNET-Pi/blob/6d64783178657dde6a157c6fbfd8cd9273546deb/scripts/birdnet_analysis.sh#L178) so I'm preparing myself for the possibility that we might find something which causes us to scratch our heads.

Let's see . . . 

Using "Tools" > "File Manager" -- Navigate to the file `/home/[yourusername]/BirdNET-Pi/scripts/birdnet_analysis.sh

At the very bottom you'll see a section that looks like this:
```
YESTERDAY="$RECS_DIR/$(date --date="yesterday" "+%B-%Y/%d-%A")"
TODAY="$RECS_DIR/$(date "+%B-%Y/%d-%A")"
if [ $(find ${YESTERDAY} -name '*wav' 2>/dev/null | wc -l) -gt 0 ];then
  find $YESTERDAY -name '*wav' -type f -size 0 -delete
  run_birdnet "${YESTERDAY}"
elif [ $(find ${TODAY} -name '*wav' | wc -l) -gt 0 ];then
  find $TODAY -name '*wav' -type f -size 0 -delete
  run_birdnet "${TODAY}"
fi
```

Let's just temporarily add a little bit to that to ask the system to look for Saturday.

Change it to look like this:
```
SATURDAY="$RECS_DIR/$(date --date="2022-09-10" "+%B-%Y/%d-%A")"
YESTERDAY="$RECS_DIR/$(date --date="yesterday" "+%B-%Y/%d-%A")"
TODAY="$RECS_DIR/$(date "+%B-%Y/%d-%A")"
if [ $(find ${SATURDAY} -name '*wav' 2>/dev/null | wc -l) -gt 0 ];then
  find $SATURDAY -name '*wav' -type f -size 0 -delete
  run_birdnet "${SATURDAY}"
elif [ $(find ${YESTERDAY} -name '*wav' | wc -l) -gt 0 ];then
  find $YESTERDAY -name '*wav' -type f -size 0 -delete
  run_birdnet "${YESTERDAY}"
elif [ $(find ${TODAY} -name '*wav' | wc -l) -gt 0 ];then
  find $TODAY -name '*wav' -type f -size 0 -delete
  run_birdnet "${TODAY}"
fi
```

The new lines:
```
SATURDAY="$RECS_DIR/$(date --date="2022-09-10" "+%B-%Y/%d-%A")"
```
Will point to Saturday
and
```
if [ $(find ${SATURDAY} -name '*wav' 2>/dev/null | wc -l) -gt 0 ];then
  find $SATURDAY -name '*wav' -type f -size 0 -delete
  run_birdnet "${SATURDAY}"
...
```
will take a whack at analyzing whatever it can find in the directory for Saturday.

:crossed_fingers:

**Reply by phillprice** [reply]
This worked perfectly and then continued on to newer days after it worked through. It's a shame previous day charts aren't generated but I'm happy with the disk space recovered :)
